File: CPP/ByRound/1728/1728_D_Letter_Picking.cpp
// Problem: CF 1728 D - Letter Picking
// https://codeforces.com/contest/1728/problem/D

/*
Purpose: Solve the letter picking game where Alice and Bob alternate picks from either end of a string,
         aiming to form lexicographically smallest strings. The game result depends on optimal play.
         Alice starts first. The winner is determined by comparing the final strings lexicographically.
         If strings are equal, it's a draw.

Algorithms/Techniques:
- Greedy strategy simulation
- Two-pointer technique to compare from both ends
- Special handling for equal characters at both ends
- Detecting if all remaining characters are the same

Time Complexity: O(n) per test case, where n is the length of the string.
Space Complexity: O(1) - only using a few variables, not counting input storage.
*/

#include <iostream>
#include <string>

using namespace std;
int main() {
  int t;
  cin >> t;
  while (t--) {
    string s;
    cin >> s;
    int l = 0, r = s.length() - 1;
    // Move pointers inward while characters at both ends are equal
    while (s[l] == s[r])
      l++, r--;
    bool dr = true; // Assume a draw initially
    // Check if all remaining characters are the same
    for (int i = l; i < r; i++, i++) // Increment by 2 for efficiency
      if (s[i] != s[i + 1])
        dr = false;
    // Output result: if all remaining are the same, it's a draw; otherwise, Alice wins
    cout << (dr ? "Draw" : "Alice") << endl;
  }
}


// https://github.com/VaHiX/CodeForces/

File: Python/ByRound/1728/1728_D_Letter_Picking.py
# Problem: CF 1728 D - Letter Picking
# https://codeforces.com/contest/1728/problem/D

"""
Algorithm: Game Theory, String Manipulation
Time Complexity: O(n^2) in worst case due to string slicing and comparison operations
Space Complexity: O(n) for storing the string and intermediate results

The problem involves a two-player game where players pick letters from either end of a string,
trying to form the lexicographically smaller string. Both players play optimally.
The key insight is to simulate the game using greedy strategy: 
- If the first and last characters of the remaining string are equal, we can remove them 
  from both ends without affecting the outcome.
- Then, we evaluate whether Alice or Bob wins by checking character pairs.

The approach checks for equal characters at both ends, removes them iteratively,
and evaluates the resulting string to determine the winner.
"""

for _ in range(int(input())):
    l = input()
    # Remove equal characters from both ends iteratively
    while l and l[-1] == l[0]:
        l = l[1:-1]
    # Check if all characters in even positions are equal
    m = 1
    for i in range(0, len(l), 2):
        m *= l[i] == l[i + 1]
    # Print result based on the comparison
    print(["Alice", "Draw"][m])


# https://github.com/VaHiX/CodeForces/